mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -600,6 +600,11 @@ LogicalResult tosa::AvgPool2dOp::verify() {
   if (inputETy.isF32() && !accType.isF32())
     return emitOpError("accumulator type for f32 tensor is not f32");
 
+  if ((llvm::isa<Float8E5M2Type>(inputETy) ||
```

Comment:
Should we move the checks in the validation pass? Maybe we can leave the operator verifier a bit more relaxed ie just check that the accumulator is of fp type?
Thoughts?
Cc: @lhutton1 @FranklandJack

Reply:
sounds good, thanks for the review. i will then close this PR.